### PR TITLE
publish speclet

### PR DESCRIPTION
**Pull request by BillWagner**
Publish the collection expressions speclet, add paragraph in the What's new article.

Contributes to #36356

See https://github.com/dotnet/docs/issues/36356#issuecomment-1680861171

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/whats-new/csharp-12.md](https://github.com/dotnet/docs/blob/b636a7898b46433f781cb63ac71dc3d1869703cc/docs/csharp/whats-new/csharp-12.md) | [What's new in C# 12](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12?branch=pr-en-us-36727) |

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and reopening for a fresh build

File: docs/csharp/whats-new/csharp-12.md [hunk 1]
```diff
@@ -21,6 +23,64 @@ The compiler generates public properties for primary constructor parameters only
...
+int[] row2 = [7, 8, 9];
+int[] single = [..row0, ..row1, ..row2];
+foreach (var element in single)
+{
+    Console.Write($"{element}, ");
+}
+// output:
+// 1, 2, 3, 4, 5, 6, 7, 8, 9,
+
+int[] vector = [..a[0..^2]];
```
Comment (tdykstra) [inline docs/csharp/whats-new/csharp-12.md:70]: The preceding example is easy to follow, but this one I have trouble parsing. You could consider adding some text explaining it. Or omit it if the preceding example adequately illustrates how the spread operator works.

File: docs/csharp/whats-new/csharp-12.md [hunk 2]
```diff
@@ -21,6 +23,64 @@ The compiler generates public properties for primary constructor parameters only
...
+    Console.Write($"{element}, ");
+}
+// output:
+// 1, 2, 3, 4, 5, 6,
+
+```
+
+The operand of a spread operator is an expression that can be enumerated. The spread operator evaluates each element of the enumerations expression.
+
+You can use collection expressions anywhere you need a collection of elements. They can specify the initial value for a collection, or be passed as arguments to methods that take collection types. You can learn more about collection expressions in the [feature specification](~/_csharplang/proposals/csharp-12.0/collection-expressions.md).
```
Comment (tdykstra) [inline docs/csharp/whats-new/csharp-12.md:82]:
```suggestion
You can use collection expressions anywhere you need a collection of elements. They can specify the initial value for a collection or be passed as arguments to methods that take collection types. You can learn more about collection expressions in the [feature specification](~/_csharplang/proposals/csharp-12.0/collection-expressions.md).
```

**Review by tdykstra — APPROVED** [review]
Looks good. I left a couple of suggestions.

File: docs/csharp/whats-new/csharp-12.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/whats-new/csharp-12.md:70]: I removed that second sample. I'll elaborate on it in the language reference updates, coming in the next PR.

**Pull request merged**